tests/test_glue_launcher.py

```diff
@@ -834,6 +834,29 @@ def test_clear_manifest_output_under_1000(self, mock_get_s3):
 
         self.assertEqual(num_cleared_files, 0)
 
+    @mock_s3
+    @mock.patch("glue_launcher_lambda.glue_launcher.get_s3_client")
+    def test_clear_manifest_output_no_files(self, mock_get_s3):
+        glue_launcher.logger = logging.getLogger()
+        bucket = "manifest_bucket"
+        prefix = (
+            "business-data/manifest/query-output_streaming_main_incremental/templates"
+        )
+        s3_client = self.setup_s3(0, bucket, prefix)
+
+        paginator = s3_client.get_paginator("list_objects_v2")
+        pages = paginator.paginate(Bucket=bucket, Prefix=prefix)
+
+        num_files = 0
+        for page in pages:
+            num_files += page["KeyCount"]
+
+        self.assertEqual(num_files, 0)
+
+        mock_get_s3.return_value = s3_client
+
+        glue_launcher.clear_manifest_output(bucket, prefix)
+
```

Comment:
What are you asserting on for the clear manifest output fn? You are simply testing the mock here

Reply:
that it doesn't error when trying to delete a prefix that is empty.

Reply:
Please assert on that then because otherwise the output of the function will be ignored and untested.
I think this test should mock does_s3_key_exist and ensure it is called with expected arg.

A second test to validate does_s3_key_exist should be created.

Reply:
I am asserting on that, if it throws an exception the test fails. 
I can assert true on response if you want? same effect. 


pointless test but sure